src/main/java/de/maxhenkel/audioplayer/command/AudioPlayerCommands.java

```diff
@@ -375,6 +381,49 @@ private static LiteralArgumentBuilder<CommandSourceStack> applyCommand(LiteralAr
+        return builder.requires((commandSource) -> commandSource.hasPermission(AudioPlayer.SERVER_CONFIG.applyToItemPermissionLevel.get()))
+                .then(Commands.argument("sound", UuidArgument.uuid())
+                        .executes((context) -> {
+                            ServerPlayer player = context.getSource().getPlayerOrException();
+                            UUID sound = UuidArgument.getUuid(context, "sound");
+                            ItemStack itemInHand = player.getItemInHand(InteractionHand.MAIN_HAND);
+                            if (validator.test(itemInHand)) {
+                                processShulker(context, itemInHand, actionItem, itemTypeName, sound, null);
+                            } else {
+                                context.getSource().sendFailure(Component.literal("You don't have a %s in your main hand".formatted(itemTypeName)));
+                            }
+                            return 1;
+                        })
+                        .then(Commands.argument("custom_name", StringArgumentType.string())
+                                .executes((context) -> {
+                                    ServerPlayer player = context.getSource().getPlayerOrException();
+                                    UUID sound = UuidArgument.getUuid(context, "sound");
+                                    ItemStack itemInHand = player.getItemInHand(InteractionHand.MAIN_HAND);
+                                    String customName = StringArgumentType.getString(context, "custom_name");
+                                    if (validator.test(itemInHand)) {
+                                        processShulker(context, itemInHand, actionItem, itemTypeName, sound, customName);
+                                    } else {
+                                        context.getSource().sendFailure(Component.literal("You don't have a %s in your main hand".formatted(itemTypeName)));
+                                    }
+                                    return 1;
+                                })));
+    }
+
+    private static void processShulker(CommandContext<CommandSourceStack> context, ItemStack itemInHand, String actionItem, String itemTypeName, UUID soundID, @Nullable String name) {
+        ListTag shulkerContents = Objects.requireNonNull(itemInHand.getTagElement(BlockItem.BLOCK_ENTITY_TAG)).getList(ShulkerBoxBlockEntity.ITEMS_TAG, Tag.TAG_COMPOUND);
+        for (int i = 0; i < shulkerContents.size(); i++) {
+            CompoundTag currentItem = shulkerContents.getCompound(i);
+            ItemStack itemStack = ItemStack.of(currentItem);
+            if ((itemStack.getItem() instanceof RecordItem && actionItem.equals("musicdisc")) || (itemStack.getItem() instanceof InstrumentItem && actionItem.equals("goathorn"))) {
+                renameItem(context, itemStack, soundID, name);
+                shulkerContents.getCompound(i).put("tag", itemStack.getTag());
```

Comment:
Similar issue here as with the other "possible null" warnings. Two choices:
1. `assert itemStack.getTag() != null;` on the line before
2. Change `.getTag()` to `.getOrCreateTag()`

Is there a preference?

Reply:
New shulker boxes don't have a tag. asserting yould crash the game. You need to use `getOrCreate` I think its called

Reply:
`Objects.requireNonNull` throws a nullpointer exception. `getOrCreateTag` would be the proper solution

Reply:
This specific line deals with the item that just got a new custom tag with the sound data, so it has to exist. Will go with `getOrCreateTag` to cover the bases

Reply:
As for the other two locations where we are dealing with the shulker box tags, I will change those to `getOrCreateTagElement` rather than the `Objects.requireNonNull`

Reply:
Done. Should all be resolved now.